virtualcluster/pkg/util/cluster/cluster.go

```diff
@@ -325,5 +325,8 @@ func (c *Cluster) SetKey(k string) {
 
 // Stop cancel/close the cache to terminate informers.
 func (c *Cluster) Stop() {
+	if c.cancelContext == nil {
```

Comment:
The cancelContext filed is initialized at line 134. Why do we do a nil check here?

Reply:
Yes, it makes no sense to panic. But it did happen, when getting VC no found. So I thought that if I make a judgment in front, it is better to do a protection during execution.

Reply:
Of course, if you think you don't need this nil check, I will close this PR. The probability of this panic happening is very low, I encountered it by chance.

Reply:
No, I never mean to close it. A crash happens means there is a bug. I am just trying to understand why it happens.  We should investigate the root cause before offering a fix.

Reply:
@m-messiah , you added the cancelFunc in the cluster struct recently, do we really need it? It seems does nothing.

Reply:
My intent of the PR https://github.com/kubernetes-sigs/cluster-api-provider-nested/pull/309/files#diff-199f4a7fe1ae9fb6f26f2b6e1e883afb86845c4b6f0a171d9daa02209160e7d6R134 was to fix the issue @wondywang is referring to. 

So, there are cases, when cancelContext is nil during runtime, so the cluster could not be stopped. 
The one solution is to add this non-nil check, but in my solution, I just initialised it as an empty func during cluster initialization to fix the initial cause.

@wondywang are you sure you use the latest code with the fix mentioned?

If yes, can you investigate a little more to catch the case where cancelContext becomes nil, please?

Reply:
I'm so sorry, @Fei-Guo @m-messiah. I looked it up, this panic happened on 13 Oct, and the latest code was merged on 25 Oct. Messiah has resolved this panic.